pp.c

```diff
@@ -2669,7 +2669,8 @@ S_negate_string(pTHX)
     }
     else return FALSE;
     SvSETMAGIC(TARG);
-    rpp_replace_1_1(TARG);
+    if (targ != sv)
```

Comment:
I think this and the others below could use:
```
    if (LIKELY(targ != sv))
```
since I expect:
```
$x = -$x;
```
to be relatively uncommon vs:
```
  foo(-$x);
```
though I could be convinced otherwise with measurements.

Reply:
Yeah, that's fair enough. I'll make those changes.